### PR TITLE
blog: survey 2023 results

### PR DESCRIPTION
**Pull request by NiklasMerz**
Announcement and deep dive into the data. Please review and add your comments.

### CODE REVIEW

File: www/_posts/2024-01-05-survey-results.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+The top 3 reasons for picking Cordova are **ease of use, cross-platform support and the prevalence and familiarity of web skills and technologies**. These reasons all were mentioned almost equally. Quite a few (~20) users mentioned that Cordova was chosen a while ago, or they migrated from Phonegap (~7). There are probably quite a few long-running Cordova projects out there. Other mentioned reasons are the support of native features, Ionic, fast development, maturity of the project and open source.
+
+> It is easy to write, inspect and debug like a webpage
+
+## Alternatives
+
+Cordova's unofficial goal has always been to cease to exist once the Web catches up with supporting new APIs and features similar to native. Therefore, it's good news that a small number of users (6.1%) migrated their apps to **PWAs**. About 13% migrated to **Capacitor** and 3% to Flutter and React-Native each. About 2% of projects moved to Electron or fully native. Around 4% are thinking about or are in the process of migrating.
+
+![Chart migrating]({{ site.baseurl }}/static/img/blog/2024/survey/image2png)
```
Comment (breautek) [inline www/_posts/2024-01-05-survey-results.md:28]:
```suggestion
![Chart migrating]({{ site.baseurl }}/static/img/blog/2024/survey/image2.png)
```

missing the dot for the file extension.

File: www/_posts/2024-01-05-survey-results.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+## Areas of improvement
+
+One of the main reasons to run this survey was to identify the pain points for many Cordova developers. If we can identify the areas that need improvements we can gather ideas for quick fixes and long term goals.
+
+The most mentioned issues is the **lack of maintenance for third party plugins**. More on that in the next paragraph. The second-biggest pain point is keeping up with App/Play store requirements and OS API changes combined with **slow releases and breaking updates**.
+
+Many users also mentioned issues with **native projects**. They tend to break with updates or require manual changes that cause headaches with Cordovas hooks and ephemeral native projects.
+
+Quite a lot of pain points also evolve around **documentation**. We should work on better documentation for config.xml and how to develop plugins. Some users also would like (video) tutorials. Sometimes information about new features or changes are hidden somewhere in PRs, issues and comments. We should do a better job updating the documentation.
```
Comment (breautek) [inline www/_posts/2024-01-05-survey-results.md:58]:
```suggestion
Quite a lot of pain points also involve around **documentation**. We should work on better documentation for config.xml and how to develop plugins. Some users also would like (video) tutorials. Sometimes information about new features or changes are hidden somewhere in PRs, issues and comments. We should do a better job updating the documentation.
```

I think the word intended here is `involve`

> evolve: develop gradually, especially from a simple to a more complex form.
> involve: (of a situation or event) include (something) as a necessary part or result.

File: www/_posts/2024-01-05-survey-results.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+Quite a lot of pain points also involve around **documentation**. We should work on better documentation for config.xml and how to develop plugins. Some users also would like (video) tutorials. Sometimes information about new features or changes are hidden somewhere in PRs, issues and comments. We should do a better job updating the documentation.
+
+Some users also fear the longevity of the project and think that new APIs and changes take too long to be supported. The project is quite old, and some code is considered legacy code today and some users would like to see support for new languages like Typescript, Swift, Kotlin.
+
+## Plugins
+
+> I’m so grateful for the plugins! I understand how hard they can be to maintain.
+
+Most respondents didn't share any feedback for plugins and quite a few mentioned they were pleased with Cordovas plugin ecosystem.
```
Comment (dpogue) [inline www/_posts/2024-01-05-survey-results.md:66]:
```suggestion
Most respondents didn't share any feedback for plugins and quite a few mentioned they were pleased with Cordova's plugin ecosystem.
```

File: www/_posts/2024-01-12-survey-results.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+> Cordova makes it simple to build plugins.
+
+## Closing words
+
+This survey got over 200 responses and considering only a fraction of people who actually see the survey will respond, there are probably quite a lot of happy Cordova users and new and long-running app projects out there. We got some good feedback of the most requested improvements and biggest pain points.
+
+As you know might now most of the maintainers are volunteers and work on Cordova on their free time. Therefore, we need your help ensuring that Cordova remains a robust and reliable platform for building mobile apps. Your involvement, whether it's providing feedback, fixing bugs, or developing plugins, is essential to ensuring Cordova's continued relevance and adoption. The Cordova PMC will do it's best to make contributing as easy as possible and provide developer guidance, code reviews, documentation, and frequent releases. A large portion of bug fixes, plugins and features have been provided by dedicated individuals for many years now and will be in the future. You can contribute to Cordova's future success with your fixes, solutions, plugins and collaboration on releases.
```
Comment (erisu) [inline www/_posts/2024-01-12-survey-results.md:90]:
```suggestion
As you might know, most of the maintainers are volunteers and work on Cordova in their free time. Therefore, we need your help ensuring that Cordova remains a robust and reliable platform for building mobile apps. Your involvement, whether it's providing feedback, fixing bugs, or developing plugins, is essential to ensuring Cordova's continued relevance and adoption. The Cordova PMC will do its best to make contributing as easy as possible and provide developer guidance, code reviews, documentation, and frequent releases. A large portion of bug fixes, plugins and features have been provided by dedicated individuals for many years now and will be in the future. You can contribute to Cordova's future success with your fixes, solutions, plugins and collaboration on releases.
```

**Comment by NiklasMerz** [comment]
I will merge this tomorrow if nobody has something to add.

**Review by breautek — APPROVED** [review]

File: www/_posts/2024-01-12-survey-results.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+## Alternatives
+
+Cordova's unofficial goal has always been to cease to exist once the Web catches up with supporting new APIs and features similar to native. Therefore, it's good news that a small number of users (6.1%) migrated their apps to **PWAs**. About 13% migrated to **Capacitor** and 3% to Flutter and React-Native each. About 2% of projects moved to Electron or fully native. Around 4% are thinking about or are in the process of migrating.
+
+![Chart migrating]({{ site.baseurl }}/static/img/blog/2024/survey/image2.png)
+
+**Only 23.2% of respondents have migrated apps to alternative solutions.**
+
+The biggest reason to migrate is **not well maintained plugins** followed by **operating system and app store compatibility issues** and **slow development of the framework itself**. This is something we as the Cordova community need to address. Some users also mentioned the fear of long term support. Others had build or Gradle issues or preferred more modern frameworks with Typescript support and a better developer experience. WebView and performance issues also got mentioned a handful of times.
```
Comment (NiklasMerz) [inline www/_posts/2024-01-12-survey-results.md:32]:
```suggestion
The biggest reason to migrate is **not well maintained plugins** followed by **operating system and app store compatibility issues** and **slow development of the framework itself**. This is something we as the Cordova community needs to address. Some users also mentioned the fear of long term support. Others had build or Gradle issues or preferred more modern frameworks with Typescript support and a better developer experience. WebView and performance issues also got mentioned a handful of times.
```

**Review by NiklasMerz — COMMENTED** [review]
Final typo

**Pull request merged**